aten/src/ATen/native/cpu/RangeFactoriesKernel.cpp

```diff
@@ -13,29 +13,29 @@ namespace {
 using namespace vec256;
 
 static void linspace_kernel(TensorIterator& iter, Scalar scalar_start, Scalar scalar_end, int64_t steps) {
-  AT_DISPATCH_ALL_TYPES_AND_COMPLEX(iter.dtype(), "linspace_cpu", [&]() {
+  AT_DISPATCH_ALL_TYPES_AND_C10_COMPLEX(iter.dtype(), "linspace_cpu", [&]() {
     // step should be of double type for all integral types
     using step_t = std::conditional_t<std::is_integral<scalar_t>::value, double, scalar_t>;
     int64_t idx(0);
     scalar_t start = scalar_start.to<scalar_t>();
     scalar_t end = scalar_end.to<scalar_t>();
-    step_t step = static_cast<step_t>(end - start) / (steps - 1);
+    step_t step = static_cast<step_t>(end - start) / static_cast<step_t>(steps - 1);
```

Comment:
why did we have to add static_cast for `steps-1`?

Reply:
I think we don't need it. Let me try removing.